### PR TITLE
PHP 7 support

### PR DESCRIPTION
**Pull request by stefansaraev**
this is based on #3. 

compile tested on php 7.0, php 7.2 and php 7.3
runtime tested on 7.0. can't effectively runtime test on 7.0+ but at least examples/add-system-connection.php runs ok. 

closes #1 
closes #2 
closes #3 

@derickr ^

### CODE REVIEW

File: php_dbus.h [hunk 1]
```diff
@@ -27,7 +27,138 @@
 
 #include "Zend/zend_hash.h"
 
-#define PHP_DBUS_VERSION "0.1.0"
+#define PHP_DBUS_VERSION "0.1.2"
```
Comment (derickr) [inline php_dbus.h]: I think it would be better to make this at least ``0.2.0``?

File: dbus.c [hunk 2]
```diff
@@ -170,91 +160,69 @@ static zend_object_handlers dbus_object_handlers_dbus_array, dbus_object_handler
...
     xmlNode         *introspect_xml;
-};
-
-#define PHP_DBUS_SIGNAL_IN  1
-#define PHP_DBUS_SIGNAL_OUT 2
+DBUS_ZEND_OBJ_STRUCT_DECL_END();
 
-struct _php_dbus_signal_obj {
-	zend_object      std;
+DBUS_ZEND_OBJ_STRUCT_DECL_BEGIN(php_dbus_signal_obj)
```
Comment (derickr) [inline dbus.c]: The use of this macro, and many others, is probably no longer useful. I suppose they were introduced to make the same code work for both PHP 5 and PHP 7.

I think I would prefer it if these macros were not added and used now, but I don't mind it too much either for more complicated macros. One liners, really ought not to be macros. (Like this, and the ``REGISTER_CLASS`` ones, for example).

File: dbus.c [hunk 3]
```diff
@@ -266,36 +234,6 @@ PHP_DBUS_SETUP_TYPE_OBJ(int64,dbus_int64_t);
...
-			return;	\
-		}	\
-	} else {	\
-		if (zend_parse_method_parameters(ZEND_NUM_ARGS() TSRMLS_CC, NULL, "O", &object, dbus_ce_dbus) == FAILURE) {	\
-			RETURN_FALSE;	\
-		}	\
-	}	\
-	obj = (php_dbus_obj *) zend_object_store_get_object(object TSRMLS_CC);	\
-
-#define DBUS_CHECK_INITIALIZED(member, class_name) \
```
Comment (derickr) [inline dbus.c]: Is this macro/check no longer needed at all?

File: .travis.yml [hunk 4]
```diff
@@ -1,7 +1,6 @@
 language: php
 
 php:
-  - 5.6
   - 7.0
```
Comment (derickr) [inline .travis.yml]: Should we add 7.1 and 7.2 here too?

File: php_dbus.h (hunk 1 above)
Comment (stefansaraev) [reply php_dbus.h]: done.

File: .travis.yml (hunk 4 above)
Comment (stefansaraev) [reply .travis.yml]: done.

File: dbus.c (hunk 3 above)
Comment (stefansaraev) [reply dbus.c]: it is not used at all even in master.

File: dbus.c (hunk 2 above)
Comment (stefansaraev) [reply dbus.c]: I wanted to keep git history and autorship from the old PR as-is. there is a commit that reverts obviously unneeded compat typedefs now. I would also like to get rid of some macros as well, but we can do later if that's okay for you.

**Comment by derickr** [comment]
Sounds good, I've merged it. Thanks a lot (and @asgrim, @Maners, and @bigbes)!

I think it would be good to make a new PECL release soon, perhaps a 0.2.0beta1? It looks like I (or somebody else) needs to update the ``package.xml`` file too (files, and dependencies)

**Comment by stefansaraev** [comment]
thanks. I've seen the package.xml files but no experience with pecl (I am building debian packages only). thought it was generated by some automated tool.

**Pull request merged**